### PR TITLE
temporal 0.12.0

### PR DESCRIPTION
**Pull request by josh-berry**
Created with `brew bump-formula-pr`.

Additionally, updated the formula to include support for auto-completion in `fish`.

<details>
  <summary>release notes</summary>
  <pre>## CLI Refresh

Temporal CLI has had its internals rewritten to be more consistent and more maintainable. This includes many improvements, but also some known backwards incompatibilities from the previous 0.11.0 release.

**:boom: WARNING: There are known compatibility breaks with this release, see the sections below :boom:**

## Known Improvements

* The position of flags is no longer significant; flags can be placed anywhere in the command invocation
* `--output json` now makes sure to only output JSON to stdout (instances of invalid JSON output have been fixed)
* `--output jsonl` now exists which disables indention/newline and, in some commands, streams lines of JSON instead of producing a single array
* `--input-meta` is now supported so different encodings can be specified (e.g. Protobuf JSON)
* Log level customization added
* `env` file path can be customized
* Payload shorthand support added, meaning most JSON base64 payloads are decoded to just show their contents
* Exit codes are now non-zero in proper failing situations
* `--api-key` can now be used to set an `Authorization` header with `Bearer` prefixed token
* `--tls-cert` and `--tls-key` data now accepted in addition to just files
* `server start-dev` now uses common log framework so log level can be more easily customized
* Logs now go to stderr instead of stdout
* `server start-dev` has reduced output noise (with more reduction to come)
* `workflow describe` now has a text form instead of always JSON
* `workflow execute` now streams event table instead of waiting on completion
* `workflow execute` properly shows non-zero exit code on failure
* `workflow execute` follows continue-as-new by default

## :boom: Known Incompatibilities

**All Commands**

* Outputs in both JSON and plain text have been changed across most commands—notably, JSON keys are now camelCase
* Duration flags need trailing unit suffix (i.e. e.g. `--workflow-timeout 5` becomes `--workflow-timeout 5s`)
* `--output table` and `--output card` removed and merged into `--output text`
* Pager removed
* `TEMPORAL_CLI_SHOW_STACKS` environment variable no longer supported
* `--tls-ca-path` can no longer be a URL
* `--fields` command removed, commands now decide which fields should be present or not

**Specific Commands**

* `env` commands now expect `--env`, `-k` and `-v` flags rather than positional arguments (positional arguments will continue to work, but are now deprecated and a warning is emitted)
* `operator namespace` commands now expect a `-n` flag to specify the namespace rather than a positional argument (the positional argument will continue to work, but is now deprecated and a warning is emitted)
* `operator namespace update` no longer supports `--verbose`
* `schedule create` changed `--workflow-type` to be `--type`
* `server start-dev` with `--db-filename` no longer auto-creates directory path of 0777 dirs if not present
* `workflow` and its subcommands no longer support `--memo-file`
* `workflow execute` does not dump history by default when JSON is enabled
* `workflow execute` when using `--event-details` (née `--fields long`) shows full JSON attributes instead of wrapped partial table
* `workflow execute` no longer succeeds if workflow exists, but `--allow-existing` is present
* `workflow reset-batch` removed in favor of `workflow reset --query` with reset options (some previous reset types may not be supported)
* `workflow show` does not allow `-f` as an alias for `--follow`
* `workflow start` no longer succeeds if workflow exists, but `--allow-existing` is present
* `workflow trace` removed while being repaired (inadvertently broken in last release)

## What's Changed
* Initial CLI rewrite by @cretz in https://github.com/temporalio/cli/pull/412
* CLI Refresh: `workflow signal` command by @cretz in https://github.com/temporalio/cli/pull/432
* Implement `workflow terminate` command by @tdeebswihart in https://github.com/temporalio/cli/pull/436
* CLI Refresh: Output improvements on start-dev by @cretz in https://github.com/temporalio/cli/pull/441
* Change log-level from "off" to "never" and show enums in options help by @cretz in https://github.com/temporalio/cli/pull/442
* Implement `workflow cancel` command by @tdeebswihart in https://github.com/temporalio/cli/pull/435
* Implement activity commands by @dandavison in https://github.com/temporalio/cli/pull/445
* Override SDK-set headers for client name and others by @cretz in https://github.com/temporalio/cli/pull/450
* Workflow start delay on CLI refresh by @cretz in https://github.com/temporalio/cli/pull/452
* More env tests by @cretz in https://github.com/temporalio/cli/pull/454
* Embed tzdata on CLI refresh by @cretz in https://github.com/temporalio/cli/pull/453
* Add query command by @Sushisource in https://github.com/temporalio/cli/pull/455
* CLI Refresh: Implement `workflow reset` by @tdeebswihart in https://github.com/temporalio/cli/pull/443
* Don't run tests in verbose mode in CI by @tdeebswihart in https://github.com/temporalio/cli/pull/444
* CLI Refresh: `workflow update` command by @pdoerner in https://github.com/temporalio/cli/pull/462
* Add `workflow stack` command by @Sushisource in https://github.com/temporalio/cli/pull/457
* Add operator cluster commands by @Quinn-With-Two-Ns in https://github.com/temporalio/cli/pull/460
* Add goreleaser and Dockerfile by @cretz in https://github.com/temporalio/cli/pull/463
* Implement `workflow show` by @Sushisource in https://github.com/temporalio/cli/pull/458
* Exit nonzero on unknown commands by @Sushisource in https://github.com/temporalio/cli/pull/466
* Enhance Task Queue Describe by @Sushisource in https://github.com/temporalio/cli/pull/467
* Add more operator cluster commands by @Quinn-With-Two-Ns in https://github.com/temporalio/cli/pull/465
* Add operator search-attribute commands by @yiminc in https://github.com/temporalio/cli/pull/470
* Add more tests by @Sushisource in https://github.com/temporalio/cli/pull/472
* [CLI Refresh] Add build id commands by @antlai-temporal in https://github.com/temporalio/cli/pull/469
* CLI Refresh: `batch` commands  by @stephanos in https://github.com/temporalio/cli/pull/464
* Add support for batch resets. by @ast2023 in https://github.com/temporalio/cli/pull/473
* Log env vars which override the env file by @Sushisource in https://github.com/temporalio/cli/pull/483
* TLS Data flags by @Sushisource in https://github.com/temporalio/cli/pull/484
* Log metrics port by @Sushisource in https://github.com/temporalio/cli/pull/482
* fix `batch terminate` test by @stephanos in https://github.com/temporalio/cli/pull/488
* JSONL support by @cretz in https://github.com/temporalio/cli/pull/487
* Adding namespace commands to cli by @prathyushpv in https://github.com/temporalio/cli/pull/476
* Adding task-queue list-partition command by @prathyushpv in https://github.com/temporalio/cli/pull/491
* CLI Refresh: `workflow delete` command by @carlydf in https://github.com/temporalio/cli/pull/496
* Add --api-key and update dependencies by @cretz in https://github.com/temporalio/cli/pull/500
* Add ARM targets for CI by @Quinn-With-Two-Ns in https://github.com/temporalio/cli/pull/497
* Add `temporal workflow fix-history-json` subcommand. by @chronos-tachyon in https://github.com/temporalio/cli/pull/504
* Implement `schedule` commands by @dnr in https://github.com/temporalio/cli/pull/459
* Do not use payload shorthand on workflow show by @cretz in https://github.com/temporalio/cli/pull/505
* Implement workflow count, temporarily remove workflow trace by @cretz in https://github.com/temporalio/cli/pull/506
* Replace release workflow with goreleaser from main by @cretz in https://github.com/temporalio/cli/pull/510
* Fixes typos in the commands Markdown source by @fairlydurable in https://github.com/temporalio/cli/pull/512
* Make argument and flag use consistent across commands by @josh-berry in https://github.com/temporalio/cli/pull/511
* Allow insecure CSRF cookies by @josh-berry in https://github.com/temporalio/cli/pull/515
* Add update-id option by @Quinn-With-Two-Ns in https://github.com/temporalio/cli/pull/517
* Fix racy batch test by @josh-berry in https://github.com/temporalio/cli/pull/516
* Exit zero if help/version is explicitly requested by @josh-berry in https://github.com/temporalio/cli/pull/525
* Add workflow trace by @sebneira in https://github.com/temporalio/cli/pull/520
* Update server and UI server by @cretz in https://github.com/temporalio/cli/pull/536
* Dev server compat issues in CLI rewrite by @cretz in https://github.com/temporalio/cli/pull/535

**Full Changelog**: https://github.com/temporalio/cli/compare/v0.11.0...v0.12.0</pre>
</details>

### CODE REVIEW

**Review by fxcoudert — APPROVED** [review]

File: Formula/t/temporal.rb [hunk 1]
```diff
@@ -21,7 +21,7 @@ class Temporal < Formula
   def install
     ldflags = "-s -w -X github.com/temporalio/cli/headers.Version=#{version}"
     system "go", "build", *std_go_args(ldflags:), "./cmd/temporal"
-    generate_completions_from_executable(bin/"temporal", "completion", shells: [:bash, :zsh])
+    generate_completions_from_executable(bin/"temporal", "completion", shells: [:bash, :fish, :zsh])
```
Comment (chenrui333) [inline Formula/t/temporal.rb:24]: you can just remove the shells block now (by default we support all three shells)
```suggestion
    generate_completions_from_executable(bin/"temporal", "completion"
```

File: Formula/t/temporal.rb [hunk 2]
```diff
@@ -21,7 +21,7 @@ class Temporal < Formula
   def install
     ldflags = "-s -w -X github.com/temporalio/cli/headers.Version=#{version}"
```
Comment (chenrui333) [inline Formula/t/temporal.rb:22]:
```suggestion
    ldflags = "-s -w -X github.com/temporalio/cli/temporalcli.Version=#{version}"
```

**Comment by chenrui333** [comment]
also squash the commits into one, thanks! :)

**Comment by josh-berry** [comment]
> also squash the commits into one, thanks! :)

Will do once I confirm the ldflags fix works.

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
Thanks @josh-berry!

**Comment by josh-berry** [comment]
> Thanks @josh-berry!

Thank you @chenrui333 for the quick review and suggestions!

**Pull request merged**